src/RelayServer.ts

```diff
@@ -374,42 +394,57 @@ export class RelayServer extends EventEmitter {
                 )
             );
         }
+
+        const { feePercentage } = this.config;
+        log.debug(`RelayServer - feePercentage: ${feePercentage}`);
+
+        const feeValue: BN = calculateFeeValue(
+            feePercentage,
+            maxPossibleGas.toString()
+        );
+
+        maxPossibleGas = maxPossibleGas.add(feeValue);
+
+        const tokenAmount: BigNumber = new BigNumber(
+            req.relayRequest.request.tokenAmount
+        );
+        const gasPrice = new BigNumber(req.relayRequest.relayData.gasPrice);
+
+        const token: Token = SUPPORTED_TOKENS.find(
+            ({ name }) => name === 'tRIF'
+        ); // FIXME: the hardcoded value should be removed once tokens are configurable and token added to the request params
+        const xRate: BigNumber = await getXRateFor(token); //TODO: replace by price feeder service
+
+        const tokenAmountInNative: BigNumber = await toNativeWeiFrom({
+            ...token,
+            amount: tokenAmount,
+            xRate
+        });
+
+        const tokenAmountInGas: BigNumber =
+            tokenAmountInNative.dividedBy(gasPrice);
+        const isTokenAmountAcceptable: boolean =
+            tokenAmountInGas.isGreaterThanOrEqualTo(maxPossibleGas.toString());
+
         log.debug(
-            'RequestFees - allowForSponsoredTx ',
-            this.config.allowForSponsoredTx
+            'RequestFees - isTokenAmountAcceptable? ',
+            isTokenAmountAcceptable
         );
-        if (!this.config.allowForSponsoredTx) {
-            // we need to convert tokenAmount back into RBTC and compare its value with maxPossibleGas
-            // if the value is lower than maxPossibleGas, we should throw an error
-            // TODO: we may need add some percentage fee at some point.
-            const tokenAmountInGas = getGas(
-                getRBTCWeiFromRifWei(
-                    toBN(req.relayRequest.request.tokenAmount)
-                ),
-                toBN(req.relayRequest.relayData.gasPrice)
+
+        if (!isTokenAmountAcceptable) {
```

Comment:
Just a question: do we support sponsored transactions in this way? if so, how?

Reply:
https://github.com/rsksmart/rif-relay-server/pull/30/files/9d95dd1776e2a5f978afe4e6dad186a17fb259ea#diff-bac0832cbd615805e4d0c6c68bf32622b7c697dea4fbc64808ac63ac43313648R406

Reply:
No, I don't think so. I think that that line covers the scenario where we don't want to add extra percentage to the gas fee, but what if we don't want to check if the transaction fees are enough? i.e.: We want to relay transactions even if the user specified zero as token fees.

Reply:
However, I think we may tackle this issue in a different task, because we may probably want to tackle this in a more sophisticated way (like allowing sponsored transactions for deployments only, etc...)

Reply:
I'm still not clear on what you mean. But if you set the fee to 0 the process will just add 0 to the gas...
Could we discuss this in sync, please?